File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/plants/plants_subclasses/Nightshade.java
package studia.prog_ob.projekt2.world_simulator.organisms.plants.plants_subclasses;

import studia.prog_ob.projekt2.world_simulator.SimLogger;
import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.plants.Plant;

public class Nightshade extends Plant {
    public static final int NIGHTSHADE_SEEDING_THRESHOLD = BASE_PLANT_SEEDING_THRESHOLD;
    public static final int NIGHTSHADE_STRENGTH = 99;
    public static final int NIGHTSHADE_INITIATIVE = BASE_PLANT_INITIATIVE;
    public static final char NIGHTSHADE_SYMBOL = 'n';


    public Nightshade(MapLocation new_map_location, World new_world) {
        super(NIGHTSHADE_STRENGTH, NIGHTSHADE_INITIATIVE, new_map_location, NIGHTSHADE_SYMBOL, new_world, NIGHTSHADE_SEEDING_THRESHOLD);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return new Nightshade(location, getWorld());
    }

    @Override
    public void collision(Organism invader) {
        SimLogger.log(invader + SimLogger.NIGHTSHADE_POISONING + this);
        invader.kill_self();
    }

    @Override
    public String toString() {
        return "Nightshade " + getMap_location() + " ";
    }
}

File: sem2/po/PO-VirtualWorld-Python/simulator/organisms/plants/Nightshade.py
from simulator.organisms.plants.Plant import *
from simulator.organisms.Organism import ORGANISM_TYPES

NIGHTSHADE_SEEDING_THRESHOLD = BASE_PLANT_SEEDING_THRESHOLD
NIGHTSHADE_STRENGTH = 99
NIGHTSHADE_INITIATIVE = BASE_PLANT_INITIATIVE


class Nightshade(Plant):
    def __init__(self, new_world, new_location):
        super().__init__(new_world, new_location, NIGHTSHADE_STRENGTH, NIGHTSHADE_INITIATIVE, ORGANISM_TYPES["Nightshade"], NIGHTSHADE_SEEDING_THRESHOLD)

    def spawn_offspring(self, new_location):
        return Nightshade(self.world, new_location)

    def __str__(self):
        return "Nightshade at " + str(self.location)

    def collision(self, invader):
        self.tlog(" poisoned " + str(invader))
        invader.kill_self()